### ISSUE TITLE
Select All in bulk delete doesn't work

### ISSUE THREAD

**Issue by SCdF** [medic/medic-webapp#3646]
All docs are set to undefined and so cannot be deleted.

**Comment by SCdF** [comment]
@garethbowen over to you. Not sure if this is the right move, there is probably a better thing that could be done here.

**Comment by garethbowen** [comment]
Back to you @SCdF

**Comment by SCdF** [comment]
Problem was with linked contacts: https://github.com/medic/medic-webapp/commit/a8afb859554ab11b90148a25e0cf5cab62bb686d#diff-ae2ab03c7390b4a1da61f490703e73daL445

I'll revert this to correctly scope this back into summaries.

**Comment by SCdF** [comment]
OK here is where I am with this:
 - There was some confusion when the lineage stuff was done, where summaries (ie search results) were mapped directly, as opposed to against a `summary` property. This broke some code. Instead, selections should have `selection.doc` or `selection.summary` depending on whether it's the summary or the expanded results
 - The above broke select all (and made the code a bit confusing), however the rendering was changed to make this map correctly.
 - The rendering was also changed for lineage stuff in general, i.e. so lineages against `selection.doc` also worked right
 - I think I have reverted the right things, but I'm not completely sure. It *nearly* looks right.

@garethbowen I'd appreciate if you took  look at the PR above and generally how it renders. The only thing I can see that is wrong right now is that the lineage doesn't generate clickable links for items you get by hitting select all, where as it does if you individually select the report. However, I can't work out why this is happening as the data structures look identical. I'll dump some screenshots on you in Slack (there is PII in them).
